### PR TITLE
Bullet physics 

### PR DESCRIPTION
**Pull request by PiotrMoscicki**

### CODE REVIEW

File: PolyEngine/Engine/Src/Physics3DWorldComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	Broadphase = std::make_unique<btDbvtBroadphase>();
+	Solver = std::make_unique<btSequentialImpulseConstraintSolver>();
+	DynamicsWorld = std::make_unique<btDiscreteDynamicsWorld>(Dispatcher.get(), Broadphase.get(), Solver.get(), CollisionConfiguration.get());
+
+	// set default gravity
+	DynamicsWorld->setGravity(btVector3(config.Gravity.X, config.Gravity.Y, config.Gravity.Z));
+}
+
+//********************************************************************************************************************************************
+Poly::Physics3DWorldComponent::~Physics3DWorldComponent()
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Physics3DWorldComponent.cpp]: Are you sure class fields are destroyed in correct order?

File: PolyEngine/Engine/Src/Physics3DWorldComponent.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	/// Config for physics simulation
+	/// @see Physics3DWorldComponent
+	/// @see Trigger3DComponent
+	/// @see Rigidbody3DComponent
+	struct ENGINE_DLLEXPORT Physics3DConfig
+	{
+	public:
+		float TimeStep = 1.f / 60.f;
+		Vector Gravity = Vector(0.f, -9.81f, 0.f);
+		float TimeFactor = 1.f;
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Physics3DWorldComponent.hpp]: What is time factor? We don't really need to speedup/slowdown simulation. We can do it via timers manipulation.

File: PolyEngine/Engine/Src/Rigidbody3DComponent.hpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		STATIC,
+		DYNAMIC,
+		DYNAMIC_CUSTOM_INTERTIA,
+		_COUNT
+	};
+
+	/// This structure is used for initialization and keeping important properties of rigid body.
+	/// @see Rigidbody3DComponent
+	/// @see eRigidBody3DType
+	struct Rigidbody3DComponentTemplate
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Rigidbody3DComponent.hpp]: BaseObject

File: PolyEngine/Engine/Src/Rigidbody3DComponent.hpp [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+		void ApplyForce(const Vector& force, const Vector& relPos);
+		void ApplyGravity();
+		void ApplyImpulse(const Vector& impulse, const Vector& relPos);
+		void ApplyTorque(const Vector& torque);
+		void ApplyTorqueImpulse(const Vector& torque);
+
+		void ClearForces();
+
+		void SetGravity(const Vector& gravity);
+		const Vector& GetGravity();
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Rigidbody3DComponent.hpp]: This method should be const.

File: PolyEngine/Engine/Src/Rigidbody3DComponent.hpp [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+	private:
+		World* BodyWorld;
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/Rigidbody3DComponent.hpp]: As stated before. We shouldn't need to store World pointer in components.

File: PolyEngine/Engine/Src/Physics3DSystem.cpp [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+const Poly::ContactResult& Poly::Physics3DSystem::ContactPair(World* world, const UniqueID& firstID, const UniqueID & secondID)
+{
+		// TODO(squares): implement this
+	return ContactResult();
+}
+
+//********************************************************************************************************************************************
+const Poly::ContactResult& Poly::Physics3DSystem::Contact(World* world, const UniqueID& entityID)
+{
+		// TODO(squares): implement this
```
[the remark this answers is not in the document]
Comment (PiotrMoscicki) [reply PolyEngine/Engine/Src/Physics3DSystem.cpp]: I'm going to do this before PR merge.

**Pull request merged**